lua/telescope/mappings.lua

```diff
@@ -75,9 +77,14 @@ local get_next_id = function()
   return _mapping_key_id
 end
 
-local assign_function = function(prompt_bufnr, func)
+local assign_function = function(prompt_bufnr, func, mode, keybind)
   local func_id = get_next_id()
 
+  if type(func) == "table" then
+    func = vim.deepcopy(func)
+    func.mode = mode
+    func.keybind = keybind
+  end
```

Comment:
Hmmm stupid idea, no idea if that work, might not be even worth to pursuit.
How about we dont deepcopy the action but rather use `vim.api.nvim_buf_get_keymap(prompt_bufnr, "n")`

We then get rhs the action command as string. First int is `prompt_bufnr`, second is `func_id`. With `__TelescopeKeymapStore[prompt_bufnr][func_id]` we then can get the action.

Reply:
Idk might be a dumb idea. Just poped in my head :laughing:

Reply:
deepcopying might not be needed but I wanted to make sure to avoid side effects. Will look into it.

Reply:
might not be needed but we dont want to change the action reference and add this to the actions in `telescope.actions` module.

Another point is, and maybe tj needs to revert this. In #987 tj will remove the `vim.deepcopy` override which will probably break this.

https://github.com/nvim-telescope/telescope.nvim/pull/987/files#diff-0811ecf01c0cf6a5e4ed485cee93d1e3194003ff3794de1be96317d93e219906L1

maybe maybe not idk right now. gotta go.

Just things that popped in my head :laughing:

Reply:
Yes, the first point is why I had it. Thanks for the important pointer, I'll look at this :) until dependencies are merged to plenary will take more time anyways.

Reply:
Refactored to follow your approach. Let me know if `utils.get_registered_mappings` is ok for you like that. Func ref is untouched now. Wondering though whether upstream should incorporate Telescope's vim.deepcopy overwrite. Without it, yes, this would have broken.